hydra-tx/exe/Options.hs

```diff
@@ -0,0 +1,221 @@
+  bstr <- Atto.takeWhile1 Char.isHexDigit
+  case deserialiseFromRawBytesHex AsTxId bstr of
+    Right addr -> return addr
+    Left e -> fail $ docToString $ "Incorrect transaction id format: " <> prettyError e
+
+parseTxIxAtto :: Atto.Parser TxIx
+parseTxIxAtto = toEnum <$> Atto.decimal
+
+outputFileParser :: Parser FilePath
+outputFileParser =
+  strOption
+    ( long "output-file"
+        <> metavar "FILE"
+        <> value "tx.json"
+        <> help "Output file path to write the transaction to."
+    )
+
+utxoParser :: Parser FilePath
+utxoParser =
+  strOption
+    ( long "utxo-file"
+        <> metavar "UTXO"
+        <> help
+          ( mconcat
+              [ "Provide a path to a file containing the UTxO to deposit/recover. "
+              , "You can use cardano-cli to query the UTxO set of an address and put the json output into a file directly. "
+              , "cardano-cli query utxo --address $(cat path-to-address.addr) --mainnet --out-file utxo.json "
+              ]
+          )
+    )
+
+headIdParser :: Parser HeadId
+headIdParser =
+  UnsafeHeadId
+    <$> strOption
+      ( long "head-id"
+          <> metavar "HEAD_ID"
+          <> help "Unique Head id identifier."
+      )
+
+networkIdParser :: Parser NetworkId
+networkIdParser = pMainnet <|> fmap Testnet pTestnetMagic
+ where
+  pMainnet :: Parser NetworkId
+  pMainnet =
+    flag'
+      Mainnet
+      ( long "mainnet"
+          <> help "Use the mainnet magic id."
+      )
+
+  pTestnetMagic :: Parser NetworkMagic
+  pTestnetMagic =
+    NetworkMagic
+      <$> option
+        auto
+        ( long "testnet-magic"
+            <> metavar "NATURAL"
+            <> value 42
+            <> showDefault
+            <> completer (listCompleter ["1", "2", "42"])
+            <> help
+              "Network identifier for a testnet to connect to. We only need to \
+              \provide the magic number here. For example: '2' is the 'preview' \
+              \network. See https://book.world.dev.cardano.org/environments.html for available networks."
+        )
+
+deadlineParser :: Parser UTCTime
+deadlineParser =
+  option
+    (posixSecondsToUTCTime . fromInteger <$> auto)
+    $ long "deadline"
+      <> metavar "DEADLINE"
+      <> help
+        ( mconcat
+            [ "Provide a deadline for the deposit transaction. "
+            , "Deposit can only be included in L2 before this deadline or reclaimed after it passes. "
+            , "Deadline is expressed as seconds since the Epoch (1970-01-01 00:00 UTC) "
+            , "To easily obtain the deadline value you can use `date` command on linux. "
+            , "`date --date '5 hours' +'%s'` -> 1725021571"
+            ]
+        )
+
+lowerBoundSlotParser :: Parser SlotNo
+lowerBoundSlotParser =
+  option
+    (SlotNo . fromInteger <$> auto)
+    $ long "slot"
+      <> metavar "LOWER_BOUND_SLOT"
+      <> help
+        ( mconcat
+            [ "Provide a starting slot for the recover transaction. "
```

Comment:
Could: add a hint that the current slot of the target network can be used here and that this must be after the deadline of a deposit to recover.